tests/migration/sle12_online_migration/yast2_migration.pm

```diff
@@ -203,10 +203,9 @@ sub run {
         }
     }
     assert_screen 'yast2-migration-target';
-    send_key "alt-p";                                       # focus on the item of possible migration targets
-    send_key_until_needlematch 'migration-target-' . get_var("VERSION"), 'down', 5;
-    wait_still_screen 5;
-    send_key "alt-n";
+    wait_screen_change(sub { send_key "alt-p" }, 20);
```

Comment:
Please see my comment about this. Make it at least `wait_screen_change { send_key 'alt-p' }, 20;`
But probably better, faster and more stable to replace with an assert_screen

Reply:
I have used 'wait_screen_change(sub { send_key "alt-p" }, 20);', it same as wait_screen_change { send_key 'alt-p' }, 20;.    I think assert_screen is not a good idea, I need wait screen show completely at here. And I have verified in this way. It can work now. I hope it still can work well when merged.

Reply:
Of course. It is just that `wait_screen_change { send_key 'alt-p' }, 20;` is the common style and the software in this project should follow a common style.

But nevertheless, the `assert_screen` *is* ensuring that the screen shows completely here assuming that you design the needle accordingly. You are saying "it is not a good idea" but without a good reason I am convinced `assert_screen` is better here

Reply:
If use assert_screen, I need delete all ''yast2-migration-target'' needles and add new needle for this one. I'm worried it will impact some other cases. In this way, I just need change code, after send key wait screen change. It has the same result as the delete needles but less effort. It can save many time.

Reply:
No, it wastes test time as wait_screen_change has a timeout which introduces waiting time in every test run. You are being too careful :)

I did not ask to delete any existing needles but to include an additional assert_screen. And even if you want to change existing needles we have good tools to also show where which needle is used